### DISCUSSION TITLE
Does photoprism plan to support https?

### DISCUSSION THREAD

**Discussion by travisbikkle** [photoprism/photoprism#2085] [Ideas]
thanks for all your guys' excellent work, photoprism is a great open source app.
but i cant find any ssl configuration after read official site, issues, discussions and roadmap.
i'm running photoprism on a server back in my house and http connection worries me, 
so are u gonna let photoprism support https?

**Comment by graciousgrey** [comment]
Yes, we plan to offer a ready-to-use solution (with https, proxy, a custom domain ...) in the future.
In the meantime you need to set it up on your own. 

You find some instructions in our docs:
- https://docs.photoprism.app/getting-started/#https
- https://docs.photoprism.app/getting-started/proxies/traefik/

**Comment by djjudas21** [comment]
If you are running PhotoPrism in Kubernetes, you get SSL termination for free if you use an Ingress. I'm using the [helm chart from k8s@home](https://github.com/k8s-at-home/charts/tree/master/charts/stable/photoprism) and it has Ingress support with Let's Encrypt to automatically provision certificates for you.

**Comment by travisbikkle** [comment]
thanks, guys, i managed to proxy photoprism with nginx, here is my nginx.conf.

```
events {
    worker_connections 1024;
}
http {
  sendfile on;
  keepalive_timeout 120;
  tcp_nodelay on;

  upstream photoprism{
      server localhost:2342;
  }

  server {
    listen 2343 ssl;
    server_name xxxxxx;
    ssl_certificate certs/mycert.pem;
    ssl_certificate_key certs/mykey.pem;
    ssl_ciphers HIGH:!aNULL:!MD5;
    charset utf-8;
    client_max_body_size 500M;
    location / {
        proxy_pass http://photoprism;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $host;
        
        proxy_buffering off;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
    }
  }
}
```

**Comment by Alestrix** [comment]
Using haproxy to add a layer of TLS is also worth a look. From what I've seen it requires very little configuration (unless you want a "real" certificate, i.e., e.g. Let's Encrypt).

**Comment by centralhardware** [comment]
i use https://nginxproxymanager.com for main reverse proxy for all selfhosted services.
